paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.cc

```diff
@@ -0,0 +1,219 @@
+  std::vector<T_out> tmp_data;
+  tmp_data.reserve(param_tensor->numel());
+
+  auto length = param_tensor->numel() / scales.size();
+
+  const float* param_data = param_tensor->data<float>();
+  for (int64_t i = 0; i < param_tensor->numel(); ++i) {
+    tmp_data[i] =
+        static_cast<T_out>(std::round(param_data[i] * scales[i / length]));
+  }
+
+  auto dims = param_tensor->dims();
+  param_tensor->clear();
+  param_tensor->Resize(dims);
+
+  auto int_param_data = param_tensor->mutable_data<T_out>(CPUPlace());
+  std::copy_n(tmp_data.data(), param_tensor->numel(), int_param_data);
+}
+
+ir::Node* FindOpInput(ir::Node* op, const std::string& input_name) {
+  auto op_input_it =
+      std::find_if(op->inputs.begin(), op->inputs.end(),
+                   [&](ir::Node* node) { return node->Name() == input_name; });
+
+  PADDLE_ENFORCE_NE(
+      op_input_it, op->inputs.end(),
+      platform::errors::InvalidArgument("Not found input %s of operator %s.",
+                                        input_name, op->Op()->Type()));
+  return (*op_input_it);
+}
+
+void ConnectInput(ir::Node* conv_op, const std::string& input_name,
+                  ir::Node* int_node) {
+  conv_op->Op()->SetInput(input_name, {int_node->Name()});
+  IR_NODE_LINK_TO(int_node, conv_op);
+}
+
+bool HasBias(ir::Node* conv_op) {
+  auto input_names = conv_op->Op()->InputNames();
+  return std::find(input_names.begin(), input_names.end(), "Bias") !=
+             input_names.end() &&
+         conv_op->Op()->Input("Bias").size() > 0;
+}
+
+bool ShouldSkipConv(ir::Node* conv_op, Scope* scope, ir::Node* conv_filter) {
+  if (!platform::HasOpINT8DataType(conv_op->Op())) {
+    return true;
+  }
+
+  auto filter_var = scope->GetVar(conv_filter->Name());
+  if (filter_var->Get<LoDTensor>().dtype() != phi::DataType::FLOAT32) {
+    VLOG(4) << "Skipping convolution (id: " << conv_op->id()
+            << ") because it's a bug that it is detected again.";
+    return true;
+  }
+
+  VLOG(4) << "Not skipping convolution (id: " << conv_op->id() << ")";
+  return false;
+}
+
+void QuantizeConvFilter(Scope* scope, ir::Graph* g, ir::Node* conv_op,
+                        ir::Node* conv_filter) {
+  const auto scale_weights =
+      conv_op->Op()->GetAttrIfExists<std::vector<float>>("Scale_weights");
+
+  auto* weight_tensor =
+      scope->GetVar(conv_filter->Name())->GetMutable<LoDTensor>();
+
+  QuantizeParams<int8_t>(weight_tensor, scale_weights);
+
+  conv_op->Op()->SetAttr("Scale_weights", std::vector<float>(1, 1));
+}
+
+void QuantizeConvBias(Scope* scope, ir::Graph* g, ir::Node* conv_op) {
+  const auto bias_scales =
+      conv_op->Op()->GetAttrIfExists<std::vector<float>>("Bias_scales");
+
+  std::string conv_bias_name = conv_op->Op()->Input("Bias")[0];
+  auto* bias_tensor = scope->GetVar(conv_bias_name)->GetMutable<LoDTensor>();
+
+  QuantizeParams<int32_t>(bias_tensor, bias_scales);
+
+  conv_op->Op()->SetAttr("Bias_scales", std::vector<float>(1, 1));
+}
+
+}  // namespace
```

Comment:
These two functions are almost the same.

Reply:
ok